blocking_queue.cpp

```diff
@@ -0,0 +1,72 @@
+#include "queue.h"
+#include "blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> BCQType, *BCQPtr;
+
+#define TO_OBJECT(x) reinterpret_cast<BCQPtr>((*x))
```

Comment:
Perhaps something like `QUEUE_FROM_HANDLE` would be clearer? Personally, I dislike helper macros that can be replaced by simple one-liners, as I find it makes the code more difficult to read.